### PR TITLE
Sieve socket creation failed

### PR DESCRIPTION
**Pull request by christer77**
This PR resolve **_Sieve: Socket creation failed_** error.

The first constant is that, since the appearance of this error, the **_BLOCK SENDER_** button is always disabled.

Related screen:

BLOCK SENDE DISABLED-SIEVE-ERROR

To reproduce this error, just open message inline mode or in new tab

### CODE REVIEW

**Comment by marclaporte** [comment]
Related: https://github.com/cypht-org/cypht/pull/1331

File: modules/imap/functions.php [hunk 1]
```diff
@@ -1523,9 +1523,13 @@ function parse_sieve_config_host($host) {
     $url = parse_url($host);
     if(!isset($url['host'])) {
         $host = $url['path'];
+    } else {
+        $host = $url['host'];
     }
     $port = $url['port'] ?? '4190';
-    return [$host, $port];
+    $scheme = $url['scheme'] ?? 'tcp://';
```
Comment (Shadow243) [inline modules/imap/functions.php:1530]: I'm not sure this will work with stalwart too, try and ping me

Comment (christer77) [reply modules/imap/functions.php:1530]: I'll get back to you after the test

**Comment by Shadow243** [comment]
Related PR: https://github.com/cypht-org/cypht/pull/1199

File: modules/imap/functions.php (hunk 1 above)
Comment (Shadow243) [reply modules/imap/functions.php:1530]: https://github.com/cypht-org/cypht/issues/1153

**Pull request closed without merging**